pkg/registries/quay/quay.go

```diff
@@ -35,28 +36,70 @@ type Quay struct {
 	config *storage.QuayConfig
 }
 
-func validate(quay *storage.QuayConfig) error {
+func validate(quay *storage.QuayConfig, categories []storage.ImageIntegrationCategory) error {
 	if quay.GetEndpoint() == "" {
 		return errors.New("Quay endpoint must be specified")
 	}
+
+	if features.QuayRobotAccounts.Enabled() {
+		if len(categories) == 1 && categories[0] == storage.ImageIntegrationCategory_SCANNER {
+			// If scanner only, robot credentials doesn't work. So expect either OAuth token or nothing (public registry)
+			if quay.GetRegistryRobotCredentials() != nil {
+				return errors.New("Quay scanner integration cannot use robot credentials")
+			}
+		} else if len(categories) == 1 && categories[0] == storage.ImageIntegrationCategory_REGISTRY {
+			// If registry only, only one of OAuth token, robot credentials or neither is allowed. Error if both are provided
+			if quay.GetRegistryRobotCredentials() != nil && quay.GetOauthToken() != "" {
+				return errors.New("Quay registry integration should use robot credentials or robot credentials but not both")
+			}
+		} else {
+			// If both scanner and registry, then ensure that we don't have robot credentials by itself
+			// That implies we have to use robot creds for scanner which is not possible.
+			// Both being empty is ok as that's a public registry.
+			if quay.GetOauthToken() == "" && quay.GetRegistryRobotCredentials() != nil {
+				return errors.New("Quay scanner integration cannot use robot credentials")
+			}
+		}
+
+		// If using robot creds, check that both username and password is provided.
+		if quay.GetRegistryRobotCredentials() != nil {
+			if quay.GetRegistryRobotCredentials().GetUsername() == "" || quay.GetRegistryRobotCredentials().GetPassword() == "" {
+				return errors.New("If using Quay robot credentials, both username and password must be provided")
```

Comment:
nit: maybe phrase this like "Both username and password must be provided when using Quay robot credentials"